streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -265,7 +265,7 @@ private void verifyAllKVKeys(final StreamRunnable[] streamRunnables, final Kafka
                 public boolean conditionMet() {
                     try {
                         final StreamsMetadata metadata = streams.metadataForKey(storeName, key, new StringSerializer());
-                        if (metadata == null) {
+                        if (metadata == null || metadata.equals(StreamsMetadata.NOT_AVAILABLE)) {
```

Comment:
Yeah, I figured they had different meanings, but seems a bit magical. :)

I would agree that the case where the store name doesn't exist should throw an exception if that is considered a user error (is there a good reason to call this method when the `storeName` doesn't exist?).

Aside from that, I'm not a fan of sentinel values because you have to know about them to use them properly. If you don't, you'll happily pass around the instance until some code tries to use something inside the sentinel that finally causes an error (probably the `HostInfo` in this case). Was this option chosen because the stream re-initialising is not considered an exceptional case?